src/App.tsx

```diff
@@ -20,6 +21,7 @@ function App() {
                 <Route path="/signup" element={<SignUp />} />
                 <Route path="/profile" element={<ProtectedRoute component={<Profile />} />} />
                 <Route path="/logout" element={<ProtectedRoute component={<Logout />} />} />
+                <Route path="*" element={<PageNotFound />} />
```

Comment:
Эта проблема оказалась настоящей головоломкой. В gh-pages если билдится из main, то возникает урл /front/ , если билдится из ПР , то возникает урл front/pr-preview/pr-№ .

Reply:
>Эта проблема оказалась настоящей головоломкой.

да :)

Есть ли сейчас какой-либо способ увидеть нашу страницу 404 в превью-билде в гитхабе?

>Но не уверен, что это хорошее решение.

почему? Какие недостатки ты в нем видишь?

>Есть еще вариант заменить BrowserRouter роутером, созданным с помощью createBrowserRouter, чтобы был доступ к data API, как например хук useRouteError

про data API я не шарю. Если понадобится, то можно заменить, но только надо сперва четко разобраться зачем, какие проблемы есть с BrowserRouter и как их они решатся при помощи createBrowserRouter

Reply:
От миграции к RouterProvider решил отказаться, т.к. вот [здесь](https://github.com/remix-run/react-router/tree/dev/examples/auth-router-provider) написано буквально следующее: _The primary difference compared to how authentication was handled in BrowserRouter is that since RouterProvider decouples fetching from rendering, we can no longer rely on React context and/or hooks to get our user authentication status. We need access to this information outside of the React tree so we can use it in our route loader and action functions._

Reply:
> Есть ли сейчас какой-либо способ увидеть нашу страницу 404 в превью-билде в гитхабе?

Нет). Gh-pages мне не победить. Никакие способы не помогли решить проблему, чтобы при вводе несуществующего урла показывалась моя страница PageNotFound. Даже уже существующие страницы типа /signin /signup при перезагрузке этих урлов выдают ошибку:
![image](https://github.com/bronfood-com/front/assets/122983365/f14c7a9c-064e-47b5-a16e-401da5f529b0)

Кстати в [Vercel](https://vercel.com/docs/deployments/git/vercel-for-github) эта же проблема решается одной строчкой в конфиге. 
Проверить можно локально npm run build $ npm run prettier. Там всё работает.
Предлагаю подождать уже сервера от заказчика и там настроить 404 страницу.

Reply:
> почему? Какие недостатки ты в нем видишь?

Потому что реально на решение моей проблемы это не влияло)

Reply:
> сервера от заказчика и там настроить 404 страницу.

Сделай пожалуйста тикет в бэклоге, чтобы не забыть об этом